crates/redstone/src/helpers/trie.rs

```diff
@@ -0,0 +1,174 @@
+use crate::network::error::Error;
+use crate::FeedId;
+
+const CHAR_START: u8 = 48; // Start at ASCII b'0'
+const CHAR_END: u8 = 90; // End at ASCII b'Z'
+const CHARS_COUNT: usize = (CHAR_END - (CHAR_START - 1)) as usize;
+
+#[derive(Debug, Clone)]
+struct Node {
+    children: [usize; CHARS_COUNT],
+    end: bool,
```

Comment:
minor: end is alittle bit misleading, is_word_end for example would be slightly clearer for me.